src/libs/actions/IOU.js

```diff
@@ -173,6 +173,31 @@ function clearMoneyRequest(transactionID) {
     Onyx.set(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, null);
 }
 
+/**
+ * Update money request-related pages IOU type params
+ *
+ * @param {Object[]} routes
+ * @param {String} newIouType
+ * @param {String} tab
+ */
+function updateMoneyRequestTypeParams(routes, newIouType, tab) {
+    routes.forEach((route) => {
+        if (!route.name.startsWith('Money_Request_') && ![CONST.TAB_REQUEST.DISTANCE, CONST.TAB_REQUEST.MANUAL, CONST.TAB_REQUEST.SCAN].includes(route.name)) {
+            return;
+        }
+        const newParams = {iouType: newIouType};
+        if (route.name === 'Money_Request_Create') {
+            // Both screen and nested params are needed to properly update the nested tab navigator
+            newParams.params = {...newParams};
+            newParams.screen = tab;
+        }
+        Navigation.setParams(newParams, route.key);
+
+        // Recursively update nested money request tab params
+        updateMoneyRequestTypeParams(lodashGet(route, 'state.routes', []), newIouType, tab);
```

Comment:
On the latest commit, I need to add this code to recursively update the nested money request tab navigator params, otherwise, we will encounter this issue:

https://github.com/Expensify/App/assets/50919443/56185e17-d4b8-4fdf-83d5-4fc35c3a4cd6

Reply:
Me neither, but it exists if we log it:
<img width="591" alt="image" src="https://github.com/Expensify/App/assets/50919443/dd650e64-d028-49a5-ae53-783c9d337e5f">

and also being used [internally on the react-navigation repo code](https://github.com/react-navigation/react-navigation/blob/aa5ae177d4d6b7ea2ae36c194b6a46747c242f9b/packages/core/src/getActionFromState.tsx#L95-L103).

The `screen` and nested `params` only appear if we refresh the participant page and go back. When we press back, it will navigate specifically to the request tab (distance, scan, manual)
https://github.com/Expensify/App/blob/5e710377c42462f7fb4266677619fc29d5092c5b/src/libs/IOUUtils.ts#L14-L24
https://github.com/Expensify/App/blob/5e710377c42462f7fb4266677619fc29d5092c5b/src/libs/Navigation/linkingConfig.ts#L377-L394

compared to when we open it through FAB.
https://github.com/Expensify/App/blob/5e710377c42462f7fb4266677619fc29d5092c5b/src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js#L194

Also, all nested tab params are undefined if we open it from FAB, but it will has the param if we navigate to the specific tab, in this example is the manual tab.
<img width="594" alt="image" src="https://github.com/Expensify/App/assets/50919443/ab3a91ef-3c62-464f-bff6-7760c926173b">

Reply:
@parasharrajat don't forget to check this one

Reply:
I just found this doc that uses screen

Please check this one https://reactnavigation.org/docs/nesting-navigators/#navigating-to-a-screen-in-a-nested-navigator

Reply:
1. Open money request page
2. Enter any amount
3. Add a user to split participant
4. Refresh and go back
5. Select Scan tab
You can see the issue in 0:18-0:19. When I press the Scan tab, the tab doesn't change at all.